specifyweb/frontend/js_src/lib/components/DataModel/legacyTypes.ts

```diff
@@ -42,20 +61,8 @@ export type SpecifyResource<SCHEMA extends AnySchema> = {
    */
   /* eslint-disable @typescript-eslint/method-signature-style */
   get<
-    FIELD_NAME extends
-      | keyof CommonFields
-      | keyof SCHEMA['fields']
-      | keyof SCHEMA['toManyDependent']
-      | keyof SCHEMA['toManyIndependent']
-      | keyof SCHEMA['toOneDependent']
-      | keyof SCHEMA['toOneIndependent'],
-    VALUE extends (CommonFields &
-      IR<never> &
-      SCHEMA['fields'] &
-      SCHEMA['toManyDependent'] &
-      SCHEMA['toManyIndependent'] &
-      SCHEMA['toOneDependent'] &
-      SCHEMA['toOneIndependent'])[FIELD_NAME]
+    FIELD_NAME extends SchemaFields<SCHEMA>,
```

Comment:
Re-run `npx tsc --extendedDiagnostics` several times with and without this change
seems like this change makes typescript type-checking of entire project faster between 0s and 1s (possibly because TypeScript can cache the resolved type?)